### PR TITLE
Issue 2999: Missing apoc.es.delete

### PR DESCRIPTION
**Pull request by vga91**
Issue `https://github.com/neo4j-contrib/neo4j-apoc-procedures/issues/2999`

~Waiting for `https://github.com/neo4j-contrib/neo4j-apoc-procedures/issues/2136`,
to replace apoc.es.postRaw with method: 'DELETE' with apoc.es.delete.~

NB: Conceptually payload should be null, but we have to put `""` instead,
as the [apoc.util.Util.writePayload method](https://github.com/neo4j/apoc/blob/dev/common/src/main/java/apoc/util/Util.java#L359) has an `if (payload == null) return;`
but we need to add the `con.setDoOutput(true);`, placed right after that condition.
Otherwise, an error `Cannot write to a URLConnection if doOutput=false - call setDoOutput(true)` will be thrown, 
and being placed in core we cannot modify the code

### Other changes

-Updated docs with a `include::partial$usage/dataset-elastic.adoc[]` and removing this link (https://github.com/neo4j-examples/elastic-example),
as it refers to neo4j 4.1.

- Changed `host-or-port` parameter to `host-or-key` in every procedure, consistent with the parameter name `String hostOrKey`,
since host-or-port is confusing, as we cannot we can't just put the port number as the first parameter

### CODE REVIEW

File: docs/asciidoc/modules/ROOT/examples/generated-documentation/apoc.es.delete.adoc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+¦xref::overview/apoc.es/apoc.es.delete.adoc[apoc.es.delete icon:book[]] +
+
```
Comment (RobertoSannino) [inline docs/asciidoc/modules/ROOT/examples/generated-documentation/apoc.es.delete.adoc:2]: riguardare

File: docs/asciidoc/modules/ROOT/pages/database-integration/elasticsearch.adoc [hunk 2]
```diff
@@ -44,6 +45,11 @@ call apoc.es.get("localhost","tweets","users","1",null,null)
 call apoc.es.stats("localhost")
 ----
 
+[source,cypher]
+----
```
Comment (RobertoSannino) [inline docs/asciidoc/modules/ROOT/pages/database-integration/elasticsearch.adoc:49]: riguardare

File: docs/asciidoc/modules/ROOT/pages/overview/apoc.es/apoc.es.delete.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
+
+= apoc.es.delete
+:description: This section contains reference documentation for the apoc.es.delete procedure.
+
+label:procedure[] label:apoc-extended[]
+
+[.emphasis]
```
Comment (RobertoSannino) [inline docs/asciidoc/modules/ROOT/pages/overview/apoc.es/apoc.es.delete.adoc:7]: rivedere

File: docs/asciidoc/modules/ROOT/pages/overview/apoc.es/index.adoc [hunk 4]
```diff
@@ -36,5 +36,9 @@ apoc.es.query(host-or-port,index-or-null,type-or-null,query-or-null,payload-or-n
 
 apoc.es.stats(host-url-Key,$config) - elastic search statistics
 |label:procedure[]
+|xref::overview/apoc.es/apoc.es.delete.adoc[apoc.es.delete icon:book[]]
+
```
Comment (RobertoSannino) [inline docs/asciidoc/modules/ROOT/pages/overview/apoc.es/index.adoc:40]: rivedere

**Comment by vga91** [comment]
Commit https://github.com/vga91/neo4j-apoc-procedures/pull/428/commits/4f5222355a7188080885299387d9463d4f5126c2: cambiato `host-or-port` con  `host-or-key` in ogni procedura, consistentemente con il nome del parametro(`String hostOrKey`),
visto che host-or-port confonde, dato che non possiamo passare semplicemente il numero della porta come primo parametro

**Comment by RobertoSannino** [comment]
LGTM

**Pull request closed without merging**